browser-sign-in/app/src/main/java/com/okta/android/samples/browser_sign_in/rxClient/RxWebAuthClientImpl.java

```diff
@@ -0,0 +1,180 @@
+import io.reactivex.subjects.PublishSubject;
+
+class RxWebAuthClientImpl implements RxWebAuthClient {
+    private WeakReference<FragmentActivity> mActivity;
+    private SyncWebAuthClient mSyncAuthClient;
+    private PublishSubject<RxResult> submitResults;
+    private RxSessionClientImpl rxSessionClientImpl;
+
+    RxWebAuthClientImpl(OIDCConfig oidcConfig, OktaState oktaState, HttpConnectionFactory connectionFactory, @ColorInt
+            int customTabColor, @Nullable String[] supportedBrowsers) {
+        rxSessionClientImpl = new RxSessionClientImpl(oidcConfig, oktaState,
+                connectionFactory);
+
+        mSyncAuthClient = new SyncWebAuthClientFactory(customTabColor, supportedBrowsers).createClient(oidcConfig, oktaState,
+                connectionFactory);
+    }
+
+    @Override
+    public Single<Boolean> isInProgress() {
+        return Single.create(emitter -> emitter.onSuccess(mSyncAuthClient.isInProgress()));
+    }
+
+    @Override
+    public Completable logIn(@NonNull FragmentActivity activity, AuthenticationPayload payload) {
+        return new Completable() {
+            @Override
+            protected void subscribeActual(CompletableObserver observer) {
+                try {
+                    mSyncAuthClient.signIn(activity, payload);
+                    observer.onComplete();
+                } catch (InterruptedException e) {
+                    processLogInResult(Result.cancel());
+                    observer.onError(e);
+                }
+            }
+        };
+    }
+
+    @Override
+    public Completable signOutFromOkta(@NonNull FragmentActivity activity) {
+        return new Completable() {
+            @Override
+            protected void subscribeActual(CompletableObserver observer) {
+                try {
+                    mSyncAuthClient.signOutFromOkta(activity);
+                    observer.onComplete();
+                } catch (InterruptedException e) {
+                    processSignOutResult(Result.cancel());
+                    observer.onError(e);
+                }
+            }
+        };
+    }
+
+    @Override
+    public Observable<RxResult> registerListener(FragmentActivity activity) {
+        submitResults = PublishSubject.create();
+
+        registerActivityLifeCycle(activity);
+        mSyncAuthClient.registerCallbackIfInterrupt(activity, (result, type) -> {
+            switch (type) {
+                case SIGN_IN:
+                    processLogInResult(result);
+                    break;
+                case SIGN_OUT:
+                    processSignOutResult(result);
+                    break;
+                default:
+                    break;
+            }
+        }, Executors.newSingleThreadExecutor());
+
+        return submitResults;
+    }
+
+    @Override
+    public RxSessionClient getSessionClient() {
+        return rxSessionClientImpl;
+    }
+
+    private void registerActivityLifeCycle(@NonNull final FragmentActivity activity) {
```

Comment:
`RxWebAuthClient` looks like it can wrap around` WebAuthClient`.  I'm not certain we can make sure developers register activity lifecycle callbacks. This does look like a `WebAuthClient` implementation so we show how to wrap around webAuthClient the lifecycle callback is done for us and this code doesn't need to be copied.

Reply:
Yes it could be. But I want to show clients how to make their own okta-client using sync implementations and what their should to remember. Maybe, they want to handler activity lifecycle in their own way. I want to make our lib for PRO-clients more flexible to other changes. I agree with you that most our clients will use our own async client, but for PRO users let's allow them to do it their self.

Reply:
ok makes sense. Then they should use their own EmptyActivityLifeCycle. Since its not public.